### PR TITLE
Add Mypy and improve typing

### PR DESCRIPTION
**Pull request by Themimitoof**
This PR brings Mypy to celery-prometheus and a CI to check if typing and lint tools are happy.

Related to T103792

### CODE REVIEW

**Comment by spyladder** [comment]
:+1:

**Review by spyladder — APPROVED** [review]

**Review by gdchamal — APPROVED** [review]

File: src/celery_prometheus/prometheus_bootstep.py [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
 
 
 if celery_version.major < 5:
-
-    from celery.signals import user_preload_options
+    from celery.signals import user_preload_options  # type: ignore
 
     @user_preload_options.connect
-    def on_preload_parsed(options, **kwargs):
+    def on_preload_parsed(options, **kwargs) -> None:
```
Comment (mardiros) [inline src/celery_prometheus/prometheus_bootstep.py]: options: Mapping[str, Any]

File: src/celery_prometheus/prometheus_bootstep.py [hunk 2]
```diff
@@ -1,37 +1,33 @@
...
         prometheus_collector_addr = options.get("prometheus_collector_addr")
         app = kwargs["app"]
         attach_prometheus_registry(app, prometheus_collector_addr)
 
 
-def add_prometheus_option(app):
-
-    help = "Celery Prometheus Configureation."
+def add_prometheus_option(app) -> None:
```
Comment (mardiros) [inline src/celery_prometheus/prometheus_bootstep.py]:
```suggestion
def add_prometheus_option(app: Celery) -> None:
```

**Review by mardiros — COMMENTED** [review]
Please restore the poetry.lock fle

File: src/celery_prometheus/prometheus_bootstep.py [hunk 3]
```diff
@@ -1,37 +1,35 @@
...
 import logging
+from argparse import ArgumentParser
+from typing import Any, Mapping, Optional, cast
 
-from celery import VERSION as celery_version
+from celery import VERSION as celery_version  # type: ignore
 from celery import Celery
-from prometheus_client import REGISTRY, start_http_server
-from prometheus_client.multiprocess import MultiProcessCollector
+from prometheus_client import REGISTRY, start_http_server  # type: ignore
```
Comment (mardiros) [inline src/celery_prometheus/prometheus_bootstep.py]: Update the prometheus_client to ^0.16.0 from the dev depenencies in order to get typing.

**Review by mardiros — APPROVED** [review]

**Pull request merged**